src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,42 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type == Bulldozer.class) {
+            List<Bulldozer> bulldozers = new BulldozerProducer().get();
+            return new ArrayList<>(bulldozers);
```

Comment:
```suggestion
        if (type == Bulldozer.class) {
            List<Bulldozer> bulldozers = bulldozerProducer.get();
            return new ArrayList<>(bulldozers);
```
you can make producers a class level variable to avoid creating a new instance each time this method is invoked